File: C/SN DAS Sir Class/Array/Array Operation/Searching/binarysearch.c
#include <stdio.h>
int bsearch(int list[], int lb, int ub, int sitem)
{
    int mid, pos = -1;
    while (lb <= ub)
    {
        mid = (lb + ub) / 2;
        if (list[mid] == sitem)
            return (mid + 1);
        else if (list[mid] > sitem)
            ub = mid - 1;
        else
            lb = mid + 1;
    }
    return (pos);
}
int main()
{
    int list[100], sitem, n, i;
    printf("Enter the number of elemnts:");
    scanf("%d", &n);
    printf("Enter the elemnts:");
    for (i = 0; i < n; i++)
        scanf("%d", &list[i]);
    printf("The items are as foloows:");
    for (i = 0; i < n; i++)
        printf("list[%d] :%d\n", i, list[i]);
    printf("Enter the item to search");
    scanf("%d", &sitem);
    int b = bsearch(list, 0, n - 1, sitem);
    if (b == -1)
        printf("Element is not presnt ");
    else
        printf("Elemnt is present at %d position", b);
    return 0;
}

File: JAVA/binarysearch.java
import java.util.*;

public class binarysearch {
    public static int bsearch(int list[], int lb, int ub, int sitem) {
        int mid, pos = -1;
        while (lb <= ub) {
            mid = (lb + ub) / 2;
            if (list[mid] == sitem)
                return (mid +1);
            else if (list[mid] > sitem)
                ub = mid - 1;
            else
                lb = mid + 1;
        }
        return (pos);
    }

    public static void main(String args[]) {
        int sitem, n, i;
        int[] list = new int[100];
        Scanner in = new Scanner(System.in);
        System.out.println("Enter the elemnts ");
        n = in.nextInt();
        System.out.println("Enter the elemnets");
        for (i = 0; i < n; i++)
            list[i] = in.nextInt();
        System.out.println("The items are as follows");
        for (i = 0; i < n; i++)
            System.out.println("list" + "[" + i + "]" + ":" + list[i]);
        System.out.println("Enter the item to search");
        sitem = in.nextInt();
        int b = bsearch(list, 0, n - 1, sitem);
        if (b == -1)
            System.out.println("Element " + sitem + " is not present");
        else
            System.out.println("Element " + sitem + " is present at position" + b);
    }
}
//Code By Rudra